### PR TITLE
Parse CSP and remove X-Frame-Options if CSP has frame-ancestors directive

### PR DESCRIPTION
**Pull request by santoru**

### CODE REVIEW

File: shcheck/shcheck.py [hunk 1]
```diff
@@ -308,6 +320,11 @@ def main():
         json_results["present"] = {}
         json_results["missing"] = []
 
+        # Before parsing, remove X-Frame-Options if there's CSP with frame-ancestors directive
+        if "frame-ancestors" in headers.get('Content-Security-Policy'.lower()).lower():
+            sec_headers.pop("X-Frame-Options")
+            headers.pop("X-Frame-Options".lower())
```
Comment (tscz) [inline shcheck/shcheck.py:326]: @santoru : I have the problem that I have CSP with frame-ancestors directive configured but no `X-Frame-Options` header. That should be a valid scenario, I guess. But this raises A KeyError: 
image

Comment (santoru) [reply shcheck/shcheck.py:326]: Apologies, should be fixed in new version (v1.6.4). Can you please confirm?

Comment (tscz) [reply shcheck/shcheck.py:326]: @santoru : perfect, works! Thanks for this super fast fix!

Comment (santoru) [reply shcheck/shcheck.py:326]: Thank you!

**Pull request merged**